example_school_library_decorator/app.rb

```diff
@@ -0,0 +1,118 @@
+require_relative './book'
+require_relative './person'
+require_relative './student'
+require_relative './teacher'
+require_relative './rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  # List all books
+  def list_all_books
+    if @books.length.positive?
+      @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+
+    else
+      puts 'There are no books at the moment'
+    end
+  end
+
+  # List all people
+  def list_all_people
+    if @people.length.positive?
+      @people.each do |person|
+        puts "name: #{person.name}, age: #{person.age}, id: #{person.id}"
+      end
+    else
+      puts 'Sorry! cannot find person'
+    end
+  end
+
+  # Create a person
+  def create_person
+    validate_person
+    @people << Student.new(@age, @name, @parent_permission)
+    @people << Teacher.new(@age, @name, @specialization)
+    puts 'Person created successfully'
+  end
+
+  # Validate person
+
+  def validate_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    @person_type = gets.chomp
+    if @person_type != '1' && @person_type != '2'
+      puts 'Invalid option'
+      nil
+    end
+    case @person_type
+    when '1'
+      print 'Age: '
+      @age = gets.chomp
+
+      print 'Name: '
+      @name = gets.chomp
+      print 'Has parent permission? [Y/N]: '
+      @parent_permission = gets.chomp
+      @parent_permission = @parent_permission.downcase == 'y'
+    when '2'
+      print 'Specialization: '
+      @specialization = gets.chomp
+    end
+  end
+
+  # Create a book
+  def create_book
+    validate_book
+    @books << Book.new(@title, @author)
+    puts 'Book created successfully'
+  end
+
+  def validate_book
+    print 'Title: '
+    @title = gets.chomp
+
+    print 'Author: '
+    @author = gets.chomp
+  end
+
+  # Create a rental
+  def create_rental
+    puts 'Please select a book from the folowing list'
+    @books.each_with_index do |book, index|
+      puts "#{index}) title: '#{book.title}', author: #{book.author}"
+    end
+    book_index = gets.chomp.to_i
+
+    puts 'Please select a person from the following list'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] name: #{person.name}, age: #{person.age}, id: #{person.id}"
+    end
+    person_index = gets.chomp.to_i
+
+    print 'date: '
+    date = gets.chomp
```

Comment:
- [ ] Kindly follow the best practices and the SOLID principles on this method as well. 😄 🙏 
>Please refer to the above request for more details.